middleware/sync.js

```diff
@@ -23,11 +23,12 @@ import { activeRouteQueries$ } from '../util/routeUtils';
 import { fetchQueries } from '../util/fetchUtils';
 
 /**
- * navRenderSub lives for the entire request, but only responds to the most
- * recent routing action, so it's a module-scoped 'SerialDisposable', which
- * will take care of disposing previous subscriptions automatically
+ * We want to make sure there is always and only one data-fetching subscription
+ * open at a time, so we create an empty subscription that is scoped to the
+ * request. When a new fetch happens, this subscription will get replaced with
+ * a new subscription.
  */
-const navRenderSub = new Rx.SerialDisposable();
+let apiFetchSub = new Rx.Subscription();
```

Comment:
not here - it gets re-assigned to a new subscription each time we need to do a new fetch, so it has to be `let`